### PR TITLE
Device: Handle network info data gathering ourselves

### PR DESCRIPTION
**Pull request by NiLuJe**
i.e., we now query routes, interfaces, wireless extensions & ping ourselves, dropping the dependency on specific CLI tools altogether.

Depends on https://github.com/koreader/koreader-base/pull/1584

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/10139)

### CODE REVIEW

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 4 of 4 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/10139)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)

**Comment by NiLuJe** [comment]
I figured why stop there, and added ping support ;D.

On the upside, I went the extra mile with unprivileged ICMP socket supports (like iputil's ping). That may help on unrooted platforms, assuming the sysctl knob doesn't block its use ;p.

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/10139)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)

**Comment by NiLuJe** [comment]
Also, because I asked myself this yesterday:

This is in Device instead of NetworkMgr mainly because it predates NetworkMgr, and this *may* be platform-specific (e.g., overloaded by specific Device implementation; while Generic does *not* setup custom NetworkMgr by default).

**Comment by NiLuJe** [comment]
Here's the money shot
netinfo

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/10139)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/10139)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)

**Comment by poire-z** [comment]
Not that I bother, but on the emulator (not running as root), it crashes:
```
02/21/23-00:56:11 ERROR Device:retrieveNetworkInfo: SIOCGIWESSID ioctl: Operation not supported
02/21/23-00:56:11 DEBUG Device:ping4: unprivileged ICMP socket: Permission denied
02/21/23-00:56:11 DEBUG Device:ping4: Opening a RAW ICMP socket requires CAP_NET_RAW capabilities!
./luajit: frontend/device/generic/device.lua:844: attempt to perform arithmetic on local 'rtt' (a nil value)
stack traceback:
        frontend/device/generic/device.lua:844: in function 'retrieveNetworkInfo'
        frontend/ui/network/networklistener.lua:243: in function 'handleEvent'
```

**Comment by NiLuJe** [comment]
Oh, duh'; the usual "did my testing before the final stupid idea" ;p.

Fix ti's trivial, will do so later tonight ;).

File: frontend/device/generic/device.lua [hunk 1]
```diff
@@ -500,57 +502,360 @@ function Device:exit()
...
+                                local essid_on = iwr.u.data.flags
+                                if essid_on ~= 0 then
+                                    local token_index = bit.band(essid_on, C.IW_ENCODE_INDEX)
+                                    if token_index > 1 then
+                                        table.insert(results, T(_("SSID: \"%1\" [%2]"), ffi.string(essid), token_index))
+                                    else
+                                        table.insert(results, T(_("SSID: \"%1\""), ffi.string(essid)))
+                                    end
+                                else
+                                    table.insert(results, _("SSID: off/any"))
```
Comment (Frenzie) [inline frontend/device/generic/device.lua:805]: What exactly does `any` mean here?

Comment (NiLuJe) [reply frontend/device/generic/device.lua:805]: That's a fair question, I'm assuming that means a hidden SSID (that's the same string as iwconfig, FWIW).

Comment (Frenzie) [reply frontend/device/generic/device.lua:805]: Ah, I didn't realize it was just copying the logic more or less verbatim. (first search result with the source, probably a bit outdated)

https://github.com/HewlettPackard/wireless-tools/blob/c1074342112d8a4cdc44275b1bc15701aaf7f30b/wireless_tools/iwconfig.c#L159-L185

**Pull request merged**